src/cpp/src/text2image/models/sd3_transformer_2d_model.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/text2image/sd3_transformer_2d_model.hpp"
+
+#include "utils.hpp"
+#include "json_utils.hpp"
+
+#include <fstream>
+
+namespace ov {
+namespace genai {
+
+SD3Transformer2DModel::Config::Config(const std::filesystem::path& config_path) {
+    std::ifstream file(config_path);
+    OPENVINO_ASSERT(file.is_open(), "Failed to open ", config_path);
+
+    nlohmann::json data = nlohmann::json::parse(file);
+    using utils::read_json_param;
+
+    read_json_param(data, "sample_size", sample_size);
+    read_json_param(data, "patch_size", patch_size);
+    read_json_param(data, "in_channels", in_channels);
+    read_json_param(data, "num_layers", num_layers);
+    read_json_param(data, "attention_head_dim", attention_head_dim);
+    read_json_param(data, "num_attention_heads", num_attention_heads);
+    read_json_param(data, "joint_attention_dim", joint_attention_dim);
+    read_json_param(data, "caption_projection_dim", caption_projection_dim);
+    read_json_param(data, "pooled_projection_dim", pooled_projection_dim);
+    read_json_param(data, "out_channels", out_channels);
+    read_json_param(data, "pos_embed_max_size", pos_embed_max_size);
+
+    file.close();
+
+    // block_out_channels should be read from VAE encoder / decoder config to compute proper m_vae_scale_factor
+    std::filesystem::path vae_config_path = config_path.parent_path().parent_path() / "vae_decoder" / "config.json";
+    file.open(vae_config_path);
+    OPENVINO_ASSERT(file.is_open(), "Failed to open ", vae_config_path);
+    data = nlohmann::json::parse(file);
+    read_json_param(data, "block_out_channels", block_out_channels);
+}
+
+SD3Transformer2DModel::SD3Transformer2DModel(const std::filesystem::path& root_dir) :
+    m_config(root_dir / "config.json") {
+    m_model = utils::singleton_core().read_model((root_dir / "openvino_model.xml").string());
+
+    // compute VAE scale factor
+    m_vae_scale_factor = std::pow(2, m_config.block_out_channels.size() - 1);
+    m_patch_size = m_config.patch_size;
+}
+
+SD3Transformer2DModel::SD3Transformer2DModel(const std::filesystem::path& root_dir,
+                const std::string& device,
+                const ov::AnyMap& properties) :
+    SD3Transformer2DModel(root_dir) {
+    compile(device, properties);
+}
+
+SD3Transformer2DModel::SD3Transformer2DModel(const SD3Transformer2DModel&) = default;
+
+const SD3Transformer2DModel::Config& SD3Transformer2DModel::get_config() const {
+    return m_config;
+}
+
+// TODO:
```

Comment:
should we remove TODO and comments within a method?

Reply:
Removed this TODO since I've already supporter `reshape` method